CONTRIBUTING_JS.md

```diff
@@ -91,12 +91,34 @@ We do not go out of our way to break compatibility with platforms, but we can on
 
 Some modules do not support certain platforms, these should be easy to spot - e.g. `@libp2p/tcp` supports only Node.js and Electron because TCP does not work in web browsers.
 
+#### Files
+
+##### TypeScript
+
+We encourage the use of TypeScript for type safety and improved developer experience. When adding new files, prefer .ts for standard TypeScript files and .tsx for React components. Ensure that TypeScript configurations align with the project's existing settings. 
+
+`aegir` can help you set up a new project with an appropriate tsconfig.json.
+
+##### Case-sensitivity
```

Comment:
Maybe make a note that this is a recommendation and that at the very least it's important to maintain consistency on a project level. Because currently not all projects use kebab-case.

For example https://github.com/ipfs/ipld-explorer-components/tree/master/src/components uses PascalCase.